_episodes/11-using-arrays-transformations.md

```diff
@@ -21,31 +21,28 @@ Arrays can be sorted, de-duplicated, and manipulated in other ways in GREL expre
 
 "Monday,Tuesday,Wednesday,Thursday,Friday,Saturday,Sunday"
 
-This can be transformed into an array using the ```split``` function
+This can be transformed into an array using the ```split``` function with pipe
 ```
-value.split(",")
+value.split("|")
 ```
-This would create the array containing the days of the week:
+This would create the array containing a list of subject headings, separated by a pipe character | (as in the data)
 
-["Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday"]
+For example:
 
-This can be combined with array operations like ```sort```. For example, assuming the cell contains the same value as above, then the function
-```
-value.split(",").sort()
-```
-would result in an array containing the days of the week sorted in alphabetical order:
+"crystal structure|clozapinium|crystal structure|molecular configuration|hydrogen bonding|supramolecular assembly|Chemistry|QD1-999"
 
-["Friday","Monday","Saturday","Sunday","Thursday","Tuesday","Wednesday"]
+This subject string can be found for the title "The crystal structures of three clozapinium salts: different molecular configurations, and supramolecular assembly in one, two and three dimensions" in the original project. 
 
-To output a value from an array you can either select a specific value depending on its position in the list (with the first position treated as 'zero'). For example
+This can be combined with array operations like ```uniques```. For example, assuming the cell contains the same value as above, then the function
 ```
-value.split(",")[0]
+value.split("|").uniques()
 ```
-would extract the first value from the array created by the ```split``` function. In the above example this would be "Monday"
+
```

Comment:
Show before/after example as well as saying which value removed

Reply:
Do I need to keep pasting in the first example so many times?

Reply:
Apologies for the long comment here - I thought maybe it would be good to just explain why this is a good example for using an array and how it works. I'm hoping based on my explanation of what we are trying to get across you might decide how much of it you think it would be useful to include in the lesson and where it's useful to include the example details?

The example is chosen to demonstrate a simple way to use an array in a transformation. It does this using a function called `uniques()` which can be used to remove duplicates from an array. In this example we start with a list of subject words:
```crystal structure|clozapinium|crystal structure|molecular configuration|hydrogen bonding|supramolecular assembly|Chemistry|QD1-999```

Examining this by eye we can see it contains "crystal structure" twice. If we assume that each cell in the subject column might have duplicates in it, and in each case the subject word/phrase that is duplicated could be different, then its not practical to "fix" this problem (remove the duplicates from each) by something like a find and replace. However, we can do it using an array. It's not trivially simple, but not really complicated either - once you get the idea of the array as something you can manipulate - which is what we need to get across to the user here I think :)

To remove the repetition we can do a GREL transformation like:
```value.split("|").uniques().join("|")```
In total this transformation does three steps:

1. `split("|")` creates an array -> `["crystal structure", "clozapinium", "crystal structure", "molecular configuration", "hydrogen bonding", "supramolecular assembly", "Chemistry", "QD1-999"]`
2. `uniques()` takes the array created, and from it generates an array with any duplicates removed (so each value in the resulting array is unique) so the result is -> `["crystal structure", "clozapinium", "molecular configuration", "hydrogen bonding", "supramolecular assembly", "Chemistry", "QD1-999"]`
3. `join("|")` turns the array created by the `uniques()` command back into a string of pipe separated values which can be stored in a cell -> `crystal structure|clozapinium|molecular configuration|hydrogen bonding|supramolecular assembly|Chemistry|QD1-999`

So we go from a list with duplicated values to a list with each value just once.